src/filehash.rs

```diff
@@ -0,0 +1,104 @@
+//! Everything related to wikis filehash functionality
+
+use InputPaths;
+use error::*;
+use std::fs::{self, File};
+use std::path::{Path, PathBuf};
+use std::io::{Write, Read, BufReader, BufRead};
+use sha_1::{Sha1, Digest};
+
+pub struct Filehash {
+}
+
+impl Filehash {
+    /// Reads the file hash for the file specified by `file_str` out of `hash_file_str`
+    fn read_file_hash(hash_file_str: &str, file_str: &str) -> Option<(String)> {
+        let hash_file_res = File::open(hash_file_str);
+        if hash_file_res.is_err() {
+            return None;
+        }
+        let hash_file_reader = BufReader::new(hash_file_res.unwrap());
```

Comment:
No, since it's checked int the line above. But since your review here I know the `if let Ok(x) = ...` construct and use just this. Thank you ;)